skins/oasis/modules/LatestActivityController.class.php

```diff
@@ -89,22 +89,21 @@ public function executeIndex() {
 				$changeList[] = $item;
 			}
 			$this->changeList = $changeList;
-			if( empty($this->userName) ) {
-				$wgMemc->set($mKeyTimes, $this->changeList, 60);
+			if ( empty( $this->userName ) ) {
+				$this->wg->Memc->set( $mKeyTimes, $this->changeList, 60 );
 			}
 		}
 
 		// Cache the response in CDN and browser
-		$this->response->setCacheValidity(600);
-		
+		$this->response->setCacheValidity( 600 );
+
 		wfProfileOut( __METHOD__ );
 	}
 
-	static function onArticleSaveComplete(&$article, &$user, $text, $summary,
-		$minoredit, $watchthis, $sectionanchor, &$flags, $revision, &$status, $baseRevId) {
-		global $wgMemc, $wgLang;
-		$wgMemc->delete(wfMemcKey('mOasisLatestActivity'));
-		$wgMemc->delete(wfMemcKey('mOasisLatestActivity_times', $wgLang->getCode()));
+	static function onArticleSaveComplete( &$article, &$user, $text, $summary, $minoredit, $watchthis, $sectionanchor, &$flags, $revision, &$status, $baseRevId ) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function has 11 parameters, which is greater than the 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)